consensus/reactor_test.go

```diff
@@ -611,7 +611,7 @@ func waitForAndValidateBlock(
 		err := validateBlock(newBlock, activeVals)
 		assert.Nil(t, err)
 		for _, tx := range txs {
-			err := assertMempool(css[j].txNotifier).CheckTx(tx, nil, mempl.TxInfo{})
+			err := assertMempool(css[j].txNotifier).CheckTx(&types.CachedTx{Tx: tx}, nil, mempl.TxInfo{})
```

Comment:
wouldn't it be easier and less breaking if we just turn it into a cached tx inside the mempool?

Reply:
what do you mean just inside the mempool? if you mean just storing it as a cachedTx then returning Tx when called, I guess we have calls of `Key()` and `Hash()` even outside the mempool, and we need to make the cachedTx used outside too to cover these.

Reply:
by inside, I mean we create the cached tx inside the mempool then return it 

this just avoids breaking the api more than what is needed

Reply:
good idea 👍  31bc8eb7ab0ced2d83e08e0ae5268b1b8e32a3ae